lib/ansible/modules/system/listen_ports_facts.py

```diff
@@ -0,0 +1,316 @@
+        listening_search = re.search('[^ ]+:[0-9]+', line)
+        if listening_search:
+            splitted = line.split()
+            conns = re.search('([^ ]+):([0-9]+)', splitted[3])
+            pidstr = ''
+            if 'tcp' in splitted[0]:
+                protocol = 'tcp'
+                pidstr = splitted[6]
+            elif 'udp' in splitted[0]:
+                protocol = 'udp'
+                pidstr = splitted[5]
+            pids = re.search('([0-9]+)/(.*)', pidstr)
+            if conns and pids:
+                address = conns.group(1)
+                port = conns.group(2)
+                pid = pids.group(1)
+                name = pids.group(2)
+                result = dict(pid=int(pid), address=address, port=int(port), protocol=protocol, name=name)
+                if result not in results:
+                    results.append(result)
+            elif not pids:
+                raise EnvironmentError('Could not get the pids for the listening ports - possibly a permission issue')
+    return results
+
+def applyWhitelist(portspids, whitelist=list()):
+    processes = list()
+    for p in portspids:
+        if int(p['port']) not in whitelist and str(p['port']) not in whitelist:
+            process = dict(pid=p['pid'], address=p['address'], port=p['port'], protocol=p['protocol'], name=p['name'], stime=p['stime'], user=p['user'])
+            if process not in processes:
+                processes.append(process)
+    return processes
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            whitelist_tcp = dict(required=False, type='list', default=list()),
+            whitelist_udp = dict(required=False, type='list', default=list())
+        ),
+        supports_check_mode=False
+    )
+
+    if platform.system() != 'Linux':
+        module.fail_json(msg='This module requires Linux.')
+
+    def getPidSTime(pid):
+        ps_cmd = module.get_bin_path('ps', True)
+        rc, ps_output, stderr = module.run_command([ps_cmd, '-o', 'lstart', '-p', str(pid)])
+        stime = ''
+        if rc == 0:
+            for line in iter(ps_output.splitlines()):
+                if 'started' not in line:
+                    stime = line
+        return stime
+
+    def getPidUser(pid):
+        ps_cmd = module.get_bin_path('ps', True)
+        rc, ps_output, stderr = module.run_command([ps_cmd, '-o', 'user', '-p', str(pid)])
+        user = ''
+        if rc == 0:
+            for line in iter(ps_output.splitlines()):
+                if line != 'USER':
+                    user = line
+        return user
+
+    result = {}
+    result['changed'] = False
+    result['ansible_facts'] = dict(tcp_listen_violations=list(), udp_listen_violations=list(), tcp_listen=list(), udp_listen=list())
+
+    try:
+        netstat_cmd = module.get_bin_path('netstat', True)
+
+        # which TCP ports are listening for connections?
+        rc, output_tcp, stderr = module.run_command([netstat_cmd, '-plnt'])
```

Comment:
It will work, but `pid`, `name`, `user`, and `stime` (the last two because `pid` is needed to determine them) values will be empty, so it could be a source of frustration when the data coming back is so bare?